### PR TITLE
Remove more mentions of removed keyword `priv`

### PR DESCRIPTION
**Pull request by carols10cents**
Hi! While researching stuff for the reference and the grammar, I came across a few mentions of using the `priv` keyword that was removed in 0.11.0 (#13547, #8122, rust-lang/rfcs#26, [RFC 0026](https://github.com/rust-lang/rfcs/blob/master/text/0026-remove-priv.md)).

One occurrence is a mention in the reference, a few are in comments, and a few are marking test functions. I left the test that makes sure you can't name an ident `priv` since it's still a reserved keyword. I did a little grepping around for `priv`, priv in backticks, `Private` etc and I think the remaining instances are fine, but if anyone knows anywhere in particular I should check for any other lingering mentions of `priv`, please let me know and I would be happy to! :fallen_leaf: :ocean:

### CODE REVIEW

**Comment by carols10cents** [comment]
r? @alexcrichton 
since he did the RFC and removal?

**Comment by alexcrichton** [comment]
@bors: r+ 2fdd1b01987314c0b0fc27a3a572d452c0b29ab1 rollup

Wow I'm surprised these lasted this long, thanks!

File: src/test/run-pass/issue-4241.rs [hunk 1]
```diff
@@ -96,7 +96,7 @@ priv fn parse_response(io: @io::Reader) -> Result {
     }
 }
 
-priv fn cmd_to_string(cmd: ~[String]) -> String {
+fn cmd_to_string(cmd: ~[String]) -> String {
```
Comment (bluss) [inline src/test/run-pass/issue-4241.rs]: This file is actually ancient. `~[String]` though, that's an anachronism?!

Comment (carols10cents) [inline src/test/run-pass/issue-4241.rs]: LOL I didn't even notice that! Maybe I'll take a look around for `~`s next :)

**Pull request merged**